controllers/errorController.js

```diff
@@ -0,0 +1,65 @@
+const AppError = require("./../utils/appError");
+
+const handleCastErrorDB = (err) => {
+  const message = `Invalid ${err.path}: ${err.value}.`;
+  return new AppError(message, 400);
+};
+
+const handleDuplicateFieldsDB = (err) => {
+  const message = `Duplicate ${Object.keys(err.keyValue).map(
+    (el) => el
+  )} value: ${
+    err.keyValue.title || err.keyValue.email
+  }. Please use another value!`;
+  return new AppError(message, 400);
+};
+const handleValidationErrorDB = (err) => {
+  const errors = Object.values(err.errors).map((el) => el.message);
+  const message = `Invalid input data. ${errors.join(". ")}`;
+  return new AppError(message, 400);
+};
+
+const sendErrorDev = (err, res) => {
+  res.status(err.statusCode).json({
+    status: err.status,
+    error: err,
+    message: err.message,
+    stack: err.stack,
+  });
+};
+
+const sendErrorProd = (err, res) => {
+  // Operational, trusted error: send message to client
+  if (err.isOperational) {
+    res.status(err.statusCode).json({
+      status: err.status,
+      message: err.message,
+    });
+
+    // Programming or other unknown error: don't leak error details
+  } else {
+    // 1) Log error
+    console.error("ERROR 💥", err);
+
+    // 2) Send generic message
+    res.status(500).json({
+      status: "error",
+      message: "Something went very wrong!",
+    });
+  }
+};
+
+module.exports = (err, req, res, next) => {
+  err.statusCode = err.statusCode || 500;
+  err.status = err.status || "error";
+
+  if (process.env.NODE_ENV === "development") {
```

Comment:
You still have not justified this

Reply:
i thought i removed it, i will send same error message in dev and prod